sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -172,7 +172,11 @@ private[sql] case class JDBCRelation(
 
   // Check if JDBCRDD.compileFilter can accept input filters
   override def unhandledFilters(filters: Array[Filter]): Array[Filter] = {
-    filters.filter(JDBCRDD.compileFilter(_, JdbcDialects.get(jdbcOptions.url)).isEmpty)
+    if (jdbcOptions.pushDownPredicate) {
```

Comment:
Are you sure, that this is the only place? JDBCRDD.scanTable defines filters as all filters that may be pushed down. Probably we should use `filters -- unhandledFilters` in JdbcRelation.buildScan

Reply:
Yes, this is the only source of truth for defining handled/unhandled. The caller calls this method and push "handled" to scanTable, in this case JDBCRDD.

Reply:
No. I share your opinion actually. It is confusing here... maybe we should change the parameter names at some point.

Reply:
Indeed it's confusing. `buildScan` argument may be named `pushedFilters`, variables also, then code will be self-describing

Reply:
Indeed it's confusing. `buildScan` argument may be named `pushedFilters`, variables also, then code will be self-describing